### PR TITLE
Tests log analysis tool

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR adds tests for the log analysis tool

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

**Comment by angrybayblade** [comment]
The test failures are caused by the connection error while downloading the tendermint binaries https://github.com/valory-xyz/open-autonomy/actions/runs/4220664472/jobs/7327282422

File: autonomy/analyse/abci/logs.py [hunk 1]
```diff
@@ -1,113 +0,0 @@
-# -*- coding: utf-8 -*-
```
Comment (angrybayblade) [inline autonomy/analyse/abci/logs.py:1]: Not required anymore

File: autonomy/analyse/logs/db.py [hunk 2]
```diff
@@ -147,24 +147,6 @@ def create(self, reset: bool = False) -> "AgentLogsDB":
 
         return self
 
-    def insert_one(  # pylint: disable=too-many-arguments
```
Comment (angrybayblade) [inline autonomy/analyse/logs/db.py:150]: Not used anywhere

File: autonomy/cli/analyse.py [hunk 3]
```diff
@@ -191,12 +195,16 @@ def docstrings(ctx: Context, update: bool) -> None:
 )
 @click.option(
     "--start-time",
-    type=str,
+    type=click.DateTime(
+        formats=(TIME_FORMAT,),
+    ),
```
Comment (angrybayblade) [inline autonomy/cli/analyse.py:200]: A small improvement to parse the datetime strings

File: autonomy/cli/analyse.py [hunk 4]
```diff
@@ -263,6 +271,8 @@ def _parse_logs(  # pylint: disable=too-many-arguments
     parser = ParseLogs()
     if logs_dir is not None:
         parser.from_dir(logs_dir=Path(logs_dir))
+        if parser.n_agents == 0:
+            raise click.ClickException(f"Cannot find agent log data in {logs_dir}")
```
Comment (angrybayblade) [inline autonomy/cli/analyse.py:275]: Raise when no agent logs found

File: tests/test_autonomy/test_cli/test_analyse/test_abci/test_log_parser.py [hunk 5]
```diff
@@ -1,132 +0,0 @@
-# -*- coding: utf-8 -*-
-# ------------------------------------------------------------------------------
```
Comment (angrybayblade) [inline tests/test_autonomy/test_cli/test_analyse/test_abci/test_log_parser.py:2]: Not relevant

File: tests/test_autonomy/test_images/base.py [hunk 6]
```diff
@@ -71,3 +80,11 @@ def build_image(
...
 
         return True, output
+
+    def teardown(self) -> None:
+        """Teardown test."""
+
+        for container in self.running_containers:
+            with contextlib.suppress(APIError):
+                container.kill()
+                container.wait(timeout=30)
```
Comment (angrybayblade) [inline tests/test_autonomy/test_images/base.py:90]: Proper teardown of the docker images

Comment (angrybayblade) [reply tests/test_autonomy/test_images/base.py:90]: The test failure here https://github.com/valory-xyz/open-autonomy/actions/runs/4202934621/jobs/7294452747 was caused by the docker image not running when performing a clean up which might have caused by the agent crashing after the starting. We only care if the agent started and entered in the registration startup behaviour, we perform this check here https://github.com/valory-xyz/open-autonomy/blob/397431b265799339b5b24b6d050e9789d8d68d63/tests/test_autonomy/test_images/test_runtime.py#L127 so no need to worry if the agent crashed afterwards

**Review by 0xArdi — APPROVED** [review]

File: tests/test_autonomy/test_images/test_runtime.py [hunk 7]
```diff
@@ -136,12 +138,6 @@ def _check_for_outputs() -> bool:
         except TimeoutError:
             successful = False
 
-        tm_container.kill()
-        tm_container.wait(timeout=5)
-
-        agent_container.kill()
-        agent_container.wait(timeout=5)
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_images/test_runtime.py:143]: Where does cleanup happen now?

File: tests/test_autonomy/test_images/test_tendermint.py [hunk 8]
```diff
@@ -97,5 +101,4 @@ def _check_for_outputs() -> bool:
             except TimeoutError:
                 successful = False
 
-            tm_container.kill()
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_images/test_tendermint.py:100]: same

File: tests/test_autonomy/test_images/test_runtime.py (hunk 7 above)
Comment (angrybayblade) [reply tests/test_autonomy/test_images/test_runtime.py:143]: https://github.com/valory-xyz/open-autonomy/pull/1834#discussion_r1111573929

**Pull request merged**